django/contrib/auth/middleware.py

```diff
@@ -34,6 +38,75 @@ def process_request(self, request):
         request.auser = partial(auser, request)
 
 
+class LoginRequiredMiddleware(MiddlewareMixin):
+    """
+    Middleware that forces all views to require login by default.
+
+    Views that have login_not_required decorator will allow unauthenticated
+    requests. Otherwise, it will redirect to a login page.
+    """
+
+    login_url = None
+    permission_denied_message = ""
+    raise_exception = False
+    redirect_field_name = REDIRECT_FIELD_NAME
+
+    def process_view(self, request, view_func, view_args, view_kwargs):
+        if request.user.is_authenticated:
+            return None
+
+        if not getattr(view_func, "login_required", True):
+            return None
+
+        return self.handle_no_permission(request, view_func)
+
+    def get_login_url(self, view_func):
+        login_url = getattr(view_func, "login_url", None)
+        if hasattr(view_func, "view_class"):
+            login_url = getattr(view_func.view_class, "login_url", login_url)
+        login_url = login_url or self.login_url or settings.LOGIN_URL
```

Comment:
The `view_class` block can be removed not that we’re no longer using class-level attributes.

```suggestion
        login_url = (
        getattr(view_func, "login_url", None)
        or self.login_url
        or settings.LOGIN_URL
    )
```

Reply:
If you were using `LoginRequiredMixin` before, set the `login_url` this would now be ignored. I don't think we should ignore it or we will have to state that the `LoginRequiredMixin` is incompatible?

Reply:
I don't intuitive that we respect the `login_required` decorator attributes but not the `LoginRequiredMixin` attributes

Reply:
Updated, I also removed the `login_url` attribute on the middleware as couldn't find the extra value compared to setting the `LOGIN_URL` setting

Reply:
Oh, right. Yeah, it’s not intuitive that `LoginRequiredMixin` wouldn’t be respected. I guess we can add back class-level support for `login_url` only..?

Reply:
`redirect_field_name` as well otherwise redirect could be broken after login if it is not the default "next"?

Reply:
Added to the release notes that `LoginRequiredMixin` is not respected. If users which to use rather than the `login_required` decorator, this can be added in a feature release 👍